src/opaque.rs

```diff
@@ -983,11 +491,45 @@ impl<CS: CipherSuite> ClientLogin<CS> {
     }
 }
 
-type ClientLoginFinishResult<CS> = (
-    LoginThirdMessage<CS>,
-    Vec<u8>,
-    GenericArray<u8, ExportKeySize>,
-);
+/// Optional parameters for client login start
+pub enum ClientLoginStartParameters {
+    /// No optional parameters
+    Default,
+    /// Specifying an info field that will be sent to the server
+    WithInfo(Vec<u8>),
+    /// Specifying the info field along with idU and idS
+    WithIdentifiersAndInfo(Vec<u8>, Vec<u8>, Vec<u8>),
+}
+
+/// Contains the fields that are returned by a client login start
+pub struct ClientLoginStartResult<CS: CipherSuite> {
+    /// The message to send to the server to begin the login protocol
+    pub credential_request: LoginFirstMessage<CS>,
+    /// The state that the client must keep in order to complete the protocol
+    pub client_login_state: ClientLogin<CS>,
+}
+
+/// Optional parameters for client login finish
+pub enum ClientLoginFinishParameters {
```

Comment:
nit: this could actually implement (and use) [`core::default::Default`](https://doc.rust-lang.org/core/default/trait.Default.html)